### PR TITLE
//mix: Add node weighting support

### PR DESCRIPTION
**Pull request by sbrl**
I found that I wanted a weighting system for the `//mix` command, so I cooked up a quick hack that adds support for weightings in the `//mix` chat command :smiley_cat:

### CODE REVIEW

**Comment by sbrl** [comment]
Just added support in the `//replace` and `//replaceinverse` commands.

Thoughts?

**Comment by sfan5** [comment]
Adding the weighting to `//replace` with that syntax breaks replacing with nodenames containing spaces.
It's also unclear what to do about the seperation of `//set` and `//mix` when the "main" command is updated to allow weighted mixing.

So I'd prefer to just have it in `//mix` for now.

File: worldedit_commands/init.lua [hunk 1]
```diff
@@ -425,15 +425,25 @@ minetest.register_chatcommand("/param2", {
...
+	params = "<node1> [<weighting1>] [<node2> [<weighting2>]] ...",
 	description = "Fill the current WorldEdit region with a random mix of <node1>, ...",
 	privs = {worldedit=true},
 	func = safe_region(function(name, param)
 		local nodes = {}
 		for nodename in param:gmatch("[^%s]+") do
-			local node = get_node(name, nodename)
-			if not node then return end
-			nodes[#nodes + 1] = node
+			if tonumber(nodename) ~= nil then
```
Comment (sfan5) [inline worldedit_commands/init.lua]: should include a check for `#nodes > 0`

**Comment by sbrl** [comment]
Thanks, @sfan5! I've updated my PR.

**Comment by sfan5** [comment]
merged as b37605943bd384cc1809df7f00740cbe1e266111

**Comment by sbrl** [comment]
Thanks, @sfan5 :smiley_cat:

**Pull request closed without merging**